### PR TITLE
[DOCS] Added update procedure

### PR DESCRIPTION
**Pull request by MagdaDziadosz**
Added upgrade procedure for standalone cluster and Cluster API

### CODE REVIEW

**Review by makhov — APPROVED** [review]

File: docs/update/update-cluster-pod.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+   apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+   kind: K0sWorkerConfig
+   metadata:
+     name: docker-test-1 # new machine
+     namespace: default
+   spec:
+     version: v1.28.7+k0s.0 # new version
+   ```
+   
+4. Run the `kubectl apply -f ./path-to-file.yaml` command to update the resources.
```
Comment (ogusarenko) [inline docs/update/update-cluster-pod.md:88]: 4. Update the resources:

   ```bash
   kubectl apply -f ./<path-to-config-file>.yaml
   ```

File: docs/update/update-cluster-pod.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    ---
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    kind: K0smotronControlPlane
+    metadata:
+      name: docker-test-cp
+    spec:
+      version: v1.27.2-k0s.0
+    ```
+
+2. Change all the k0s versions to the latest one:
```
Comment (ogusarenko) [inline docs/update/update-cluster-pod.md:46]: That would be nice to add a link to the k0s describing the versioning. And/or add the command there to find out what the latest k0s version is, if it exists.

File: docs/update/update-cluster-pod.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# Update Cluster API cluster
+
+To update k0smotron cluster deployed with Cluster API, you need to update
+the k0s version and machine names in the YAML configuration file.
+
+!!! warning "Data loss" 
+
+    The described example does not include persistence.
+    If you proceed with it, you will lose all your data during the update process.
+    To prevent data loss, you can update workers manually or use [k0s autopilot](https://docs.k0sproject.io/stable/autopilot/) instead.
```
[the remark this answers is not in the document]
Comment (MagdaDziadosz) [reply docs/update/update-cluster-pod.md:10]: That is a great question for @makhov

File: docs/update/update-cluster-pod.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    metadata:
+      name: docker-test-cp
+    spec:
+      version: v1.27.2-k0s.0
+      persistence:
+        type: hostPath
+        hostPath: "/tmp/kmc-test" # k0smotron will mount a basic hostPath volume to avoid data loss.
+   ```
+
+   Do not configure persistence in production environment.
```
Comment (jnummelin) [inline docs/update/update-cluster-pod.md:55]: I think this is supposed to read like "do not configure `hostPath` persistence in production..."

File: docs/update/update-cluster-pod.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+     namespace: default
+   spec:
+     version: v1.28.7+k0s.0 # new version
+   ```
+ 
+5. Update the resources:
+
+   ```bash
+   kubectl apply -f ./path-to-file.yaml
+
```
Comment (jnummelin) [inline docs/update/update-cluster-pod.md:105]: This seems to be missing the closing code segment

File: docs/update/update-standalone.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    metadata:
+      name: docker-test-cp
+    spec:
+      version: v1.27.2-k0s.0
+      persistence:
+        type: hostPath
+        hostPath: "/tmp/kmc-test" # k0smotron will mount a basic hostPath volume to avoid data loss.
+   ```
+
+   Do not configure persistence in production environment.
```
Comment (jnummelin) [inline docs/update/update-standalone.md:35]: Again, probably meant hostPath here

**Review by jnummelin — CHANGES_REQUESTED** [review]

**Review by jnummelin — APPROVED** [review]

File: docs/update/update-cluster-pod.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
+# Update k0smotron in Cluster API integrated cluster
+
+To update k0smotron cluster deployed with Cluster API, you need to update
+the k0s version and machine names in the YAML configuration file.
```
Comment (ogusarenko) [inline docs/update/update-cluster-pod.md:4]:
```suggestion
the k0s version and machine names in the YAML configuration file:
```

File: docs/update/update-cluster-pod.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    metadata:
+      name: docker-test-cp
+    spec:
+      version: v1.27.2-k0s.0
+      persistence:
+        type: hostPath
+        hostPath: "/tmp/kmc-test" # k0smotron will mount a basic hostPath volume to avoid data loss.
+   ```
+
+   Do not configure `hostPath` persistence in production environment.
```
Comment (ogusarenko) [inline docs/update/update-cluster-pod.md:55]: Using the `hostPath` volume type introduces many security risks. Avoid configuring persistence for volumes of the `hostPath` type.

File: docs/update/update-cluster-pod.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+      name: docker-test-cp
+    spec:
+      version: v1.27.2-k0s.0
+      persistence:
+        type: hostPath
+        hostPath: "/tmp/kmc-test" # k0smotron will mount a basic hostPath volume to avoid data loss.
+   ```
+
+   Do not configure `hostPath` persistence in production environment. 
+   Learn more from the official Kubernetes documentation on [hostPath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath).
```
Comment (ogusarenko) [inline docs/update/update-cluster-pod.md:56]:
```suggestion
   Learn more from [official Kubernetes documentation: hostPath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath).
```

File: docs/update/update-cluster-pod.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+3. Change all the k0s versions to [the target one](https://docs.k0sproject.io/v1.29.2+k0s.0/releases/#k0s-release-and-support-model). For example:
```
Comment (ogusarenko) [inline docs/update/update-cluster-pod.md:58]: Remove this link from the step, not the distract the reader. You can add a see also block at the end including the ref

File: docs/update/update-cluster-pod.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+   ```bash
+   kubectl apply -f ./path-to-file.yaml
+   ```
+
+   
+6. Remove the machines running the old k0smotron version:
+
+   ```bash
+   kubectl delete machine docker-test-0
+   ```
```
Comment (ogusarenko) [inline docs/update/update-cluster-pod.md:112]: That would be great to have a summary at the end with the verification step

**Review by ogusarenko — COMMENTED** [review]
Please apply my comments from the 1st file to the 2nd file as well. Thank you!

File: docs/update/update-standalone.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+      version: v1.28.7-k0s.0 # new k0s version
+    ```
+
+4. Update the resources:
+
+   ```bash
+   kubectl apply -f ./path-to-file.yaml
+   ```
+
+The update procedure is completed, you now have the latest k0smotron version.
```
Comment (ogusarenko) [inline docs/update/update-standalone.md:58]:
```suggestion
The update procedure is completed, you now have the target version of k0smotron.
```

**Review by ogusarenko — APPROVED** [review]

**Pull request merged**